File: src/moveit_fk_demo.cpp
/***********************************************************************
# Copyright 2020 The UmbraTek Inc. All Rights Reserved.
#
# Software License Agreement (BSD License)
#
# Author: johnson huang<johnson@umbratek.com>
# 
***********************************************************************/

#include <ros/ros.h>
#include <moveit/move_group_interface/move_group_interface.h>
// api reference https://github.com/ros-planning/moveit/blob/master/moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h

int main(int argc, char **argv)
{
    ros::init(argc, argv, "utra_moveit_fk_demo");
    ros::AsyncSpinner spinner(1);
    spinner.start();

    moveit::planning_interface::MoveGroupInterface arm("manipulator");

    arm.setGoalJointTolerance(0.001);

    arm.setMaxAccelerationScalingFactor(0.1);
    arm.setMaxVelocityScalingFactor(0.1);

    //  go to home pose
    arm.setNamedTarget("home");
    arm.move();
    sleep(1);

    // go to this joints, plan first, and excute
    double targetPose_1[6] = {0, 0, 1.53, 0, 1.53, 0};
    std::vector<double> joint_group_positions_1(6);
    joint_group_positions_1[0] = targetPose_1[0];
    joint_group_positions_1[1] = targetPose_1[1];
    joint_group_positions_1[2] = targetPose_1[2];
    joint_group_positions_1[3] = targetPose_1[3];
    joint_group_positions_1[4] = targetPose_1[4];
    joint_group_positions_1[5] = targetPose_1[5];

    arm.setJointValueTarget(joint_group_positions_1);
    moveit::planning_interface::MoveGroupInterface::Plan plan_1;
    arm.plan(plan_1);
    sleep(2);
    arm.execute(plan_1);

    // go to home pose, go immediately
    double targetPose_2[6] = {0, 0, 0, 0, 0, 0};
    std::vector<double> joint_group_positions_2(6);
    joint_group_positions_2[0] = targetPose_2[0];
    joint_group_positions_2[1] = targetPose_2[1];
    joint_group_positions_2[2] = targetPose_2[2];
    joint_group_positions_2[3] = targetPose_2[3];
    joint_group_positions_2[4] = targetPose_2[4];
    joint_group_positions_2[5] = targetPose_2[5];

    arm.setJointValueTarget(joint_group_positions_2);
    moveit::planning_interface::MoveGroupInterface::Plan plan_2;
    arm.move();
    sleep(1);

    ros::shutdown(); 

    return 0;
}

File: scripts/moveit_fk_demo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# Copyright 2020 The UmbraTek Inc. All Rights Reserved.
#
# Software License Agreement (BSD License)
#
# Author: johnson huang<johnson@umbratek.com>
# =============================================================================

import rospy, sys
import moveit_commander
#api reference https://github.com/ros-planning/moveit/blob/master/moveit_commander/src/moveit_commander/move_group.py

class MoveItFkDemo:
    def __init__(self):
        moveit_commander.roscpp_initialize(sys.argv)
        rospy.init_node('utra_moveit_fk_demo', anonymous=True)
 
        self.arm = moveit_commander.MoveGroupCommander('manipulator')
        
        # set tolerance
        self.arm.set_goal_joint_tolerance(0.001)

        # set max velocity acceleration
        self.arm.set_max_acceleration_scaling_factor(0.1)
        self.arm.set_max_velocity_scaling_factor(0.1)

        # go to home pose
        self.arm.set_named_target('home')
        self.arm.go()

        # go to this joints, plan first, and excute
        joint_positions = [0, 0, 1.53, 0, 1.53, 0]
        self.arm.set_joint_value_target(joint_positions)
        plan_msg = self.arm.plan()
        rospy.sleep(2)
        self.arm.execute(plan_msg)

        # go to home pose, go immediately
        joint_positions = [0, 0, 0, 0, 0, 0]
        self.arm.set_joint_value_target(joint_positions)
        self.arm.go()


        moveit_commander.roscpp_shutdown()
        moveit_commander.os._exit(0)

    

if __name__ == "__main__":
    try:
        fk_demo = MoveItFkDemo()
       
    except rospy.ROSInterruptException:
        pass
